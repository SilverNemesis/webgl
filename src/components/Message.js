import React from 'react';

const Message = ({ message, onClick }) => {
  if (!message) {
    return null;
  }
  return (
    <div id="message" onClick={onClick}>
      {message.map((text, index) => (<div key={index}>{text}</div>))}
    </div>
  );
}

export default Message;
