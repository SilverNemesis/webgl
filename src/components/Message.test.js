import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Message from './Message'

const renderer = new ShallowRenderer();

it('renders correctly', () => {
    const message = ['This is my message', 'It is about nothing', 'But it has three lines'];
    renderer.render(<Message message={message} />)
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
});
