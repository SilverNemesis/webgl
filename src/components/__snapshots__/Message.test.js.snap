// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="message"
>
  <div>
    This is my message
  </div>
  <div>
    It is about nothing
  </div>
  <div>
    But it has three lines
  </div>
</div>
`;
